test/parallel/test-net-isip.js

```diff
@@ -24,70 +24,36 @@ require('../common');
-                   0);
-assert.strictEqual(net.isIP('1050:0:0:0:5:600:300c:326b'), 6);
-assert.strictEqual(net.isIP('2001:252:0:1::2008:6'), 6);
-assert.strictEqual(net.isIP('2001:dead:beef:1::2008:6'), 6);
-assert.strictEqual(net.isIP('2001::'), 6);
-assert.strictEqual(net.isIP('2001:dead::'), 6);
-assert.strictEqual(net.isIP('2001:dead:beef::'), 6);
-assert.strictEqual(net.isIP('2001:dead:beef:1::'), 6);
-assert.strictEqual(net.isIP('ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff'), 6);
-assert.strictEqual(net.isIP(':2001:252:0:1::2008:6:'), 0);
-assert.strictEqual(net.isIP(':2001:252:0:1::2008:6'), 0);
-assert.strictEqual(net.isIP('2001:252:0:1::2008:6:'), 0);
-assert.strictEqual(net.isIP('2001:252::1::2008:6'), 0);
-assert.strictEqual(net.isIP('::2001:252:1:2008:6'), 6);
-assert.strictEqual(net.isIP('::2001:252:1:1.1.1.1'), 6);
-assert.strictEqual(net.isIP('::2001:252:1:255.255.255.255'), 6);
-assert.strictEqual(net.isIP('::2001:252:1:255.255.255.255.76'), 0);
-assert.strictEqual(net.isIP('::anything'), 0);
-assert.strictEqual(net.isIP('::1'), 6);
-assert.strictEqual(net.isIP('::'), 6);
-assert.strictEqual(net.isIP('0000:0000:0000:0000:0000:0000:12345:0000'), 0);
-assert.strictEqual(net.isIP('0'), 0);
-assert.strictEqual(net.isIP(), 0);
-assert.strictEqual(net.isIP(''), 0);
-assert.strictEqual(net.isIP(null), 0);
-assert.strictEqual(net.isIP(123), 0);
-assert.strictEqual(net.isIP(true), 0);
-assert.strictEqual(net.isIP({}), 0);
-assert.strictEqual(net.isIP({ toString: () => '::2001:252:1:255.255.255.255' }),
-                   6);
-assert.strictEqual(net.isIP({ toString: () => '127.0.0.1' }), 4);
-assert.strictEqual(net.isIP({ toString: () => 'bla' }), 0);
+const ip = [
+  ['127.0.0.1', 4],
+  ['8.8.8.8', 4],
+  ['x127.0.0.1', 0],
+  ['example.com', 0],
+  ['0000:0000:0000:0000:0000:0000:0000:0000', 6],
+  ['0000:0000:0000:0000:0000:0000:0000:0000::0000', 0],
+  ['1050:0:0:0:5:600:300c:326b', 6],
+  ['2001:252:0:1::2008:6', 6],
+  ['2001::', 6],
+  ['2001:dead::', 6],
+  ['2001:dead:beef::', 6],
+  ['2001:dead:beef:1::', 6],
+  ['ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff', 6],
+  [':2001:252:0:1::2008:6:', 0],
+  [':2001:252:0:1::2008:6', 0],
+  ['2001:252:0:1::2008:6:', 0],
+  ['2001:252::1::2008:6', 0],
+  ['::2001:252:1:2008:6', 6],
+  ['::2001:252:1:1.1.1.1', 6],
+  ['::2001:252:1:255.255.255.255', 6],
+  ['::2001:252:1:255.255.255.255.76', 0],
+  ['::anything', 0],
+  ['::1', 6],
+  ['::', 6],
+  ['0000:0000:0000:0000:0000:0000:12345:0000', 0],
+  ['0', 0]
+];
 
-assert.strictEqual(net.isIPv4('127.0.0.1'), true);
-assert.strictEqual(net.isIPv4('example.com'), false);
-assert.strictEqual(net.isIPv4('2001:252:0:1::2008:6'), false);
-assert.strictEqual(net.isIPv4(), false);
-assert.strictEqual(net.isIPv4(''), false);
-assert.strictEqual(net.isIPv4(null), false);
-assert.strictEqual(net.isIPv4(123), false);
-assert.strictEqual(net.isIPv4(true), false);
-assert.strictEqual(net.isIPv4({}), false);
```

Comment:
These cases are not tested in the other file. Is there a specific reason for removing them?

Reply:
About this, I think `isIPxx` since it only supports string input, testing non-string input is an unnecessary test, don't get useful feedback in run the test case. 

I'm not sure if it should do this kind of testing in node,  imo, It is up to the user to make a simple assertion before using it.

Reply:
I am not certain I can follow. We verify that the API works as expected and to have a full coverage.